File: boj_codes/Python3/1002.py
#https://www.acmicpc.net/problem/1002
# 20200430

import sys

def find_connection(dist, r1, r2):
    rdiff = (r1 - r2) if r1 >= r2 else (r2 - r1)
    rdiff = rdiff ** 2
    rsum = (r1 + r2) ** 2

    if dist < rdiff:
        return 0

    elif dist == rdiff:
        if dist == 0:
            return -1
        else:
            return 1

    elif rsum > dist:
        return 2

    elif rsum == dist:
        return 1

    elif rsum < dist:
        return 0



N = int(sys.stdin.readline().rstrip())

for _ in range(N):
    x1, y1, r1, x2, y2, r2 = map(int, sys.stdin.readline().split())
    dist = ((x1 - x2) ** 2) + ((y1 - y2) ** 2)
    print(find_connection(dist, r1, r2))

File: boj_codes/C++17/1002.cpp
// https://www.acmicpc.net/problem/1002
// 20201219

#include <cstdio>

int find_intersect(int diff, int r_1, int r_2){
    int rdiff = (r_1 - r_2) * (r_1 - r_2);
    int rsum = (r_1 + r_2) * (r_1 + r_2);

    if(rdiff > diff)
        return 0;
    
    else if(rdiff == diff){
        if(diff == 0)
            return-1;
        else
            return 1;
    }

    else if(diff > rsum)
        return 0;    
        
    else if(diff == rsum)
        return 1;
    
    else if(diff < rsum)
        return 2;
    
    return 0;

}

int main(){
    int Test_case;
    int x_1, x_2, y_1, y_2, r_1, r_2;
    int diff;
    scanf("%d", &Test_case);

    while(Test_case--){
        scanf("%d %d %d %d %d %d", &x_1, &y_1, &r_1, &x_2, &y_2, &r_2);

        diff = ((x_1 - x_2) * (x_1 - x_2)) +((y_1 - y_2) * (y_1 - y_2));
        printf("%d\n", find_intersect(diff, r_1, r_2));
    }

    return 0;
}